fitbenchmarking/fitbenchmarking
# Maximum Time Option

#### Description of Work
This PR adds a `max_runtime` option which allows you to set a maximum time that you want your benchmark of each fitting combination to run for. The default has been set to 10 minutes (600 seconds) - other suggestions are welcome.

The flag 6 indicates when the `max_runtime` is reached.

Fixes #846 

#### Testing Instructions

1.  Set `max_runtime` option and check that slow minimizers are not run and instead show error flag 6.

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
